hack/lib/cmd.sh

```diff
@@ -193,23 +193,14 @@ function os::cmd::internal::expect_exit_code_run_grep() {
 
 	local test_result=0
 	if [[ -n "${grep_args}" ]]; then
-		test_result=$( os::cmd::internal::run_collecting_output 'os::cmd::internal::get_results | grep -Eq "${grep_args}"'; echo $? )
-
+		test_result=$( os::cmd::internal::run_collecting_output "grep -Eq '${grep_args}' <(os::cmd::internal::get_results)"; echo $? )
```

Comment:
They were technically correct both times since the `os::cmd::internal::run_collecting_output` does an `eval` ... but it's better practice to have the args (*e.g.* `${grep_args}`) expand in the shell before the `eval` so things like the stack trace can better signal to us what exactly was going on.